src/TaskModel.vala

```diff
@@ -640,6 +678,62 @@ public class Tasks.TaskModel : Object {
         }
     }
 
+    private async void update_gtasks_tasklist (E.Source task_list, ECal.Client client, E.SourceRegistry registry) {
```

Comment:
Since this method completely bypasses EDS and seems not to be related to sorting at all, we might want to solve the refresh problem in another PR which also takes other backends into account. EDS' refresh for a WebDAV backend does not work 100% as expected too.